src/framework/components/element/text-element.js

```diff
@@ -131,21 +179,50 @@ pc.extend(pc, function () {
                 }
                 this._entity.addChild(this._model.graph);
                 this._model._entity = this._entity;
+
+                this._updateAligns();
             } else {
                 this._updateMesh(this._mesh, text);
                 this._meshInstance.setParameter("texture_msdfMap", this._font.texture);
             }
         },
 
-        _updateMaterial: function (screenSpace) {
-            if (screenSpace) {
-                this._material = this._system.defaultScreenSpaceTextMaterial;
-                if (this._meshInstance) this._meshInstance.layer = pc.scene.LAYER_HUD;
-            } else {
-                this._material = this._system.defaultTextMaterial;
-                if (this._meshInstance) this._meshInstance.layer = pc.scene.LAYER_WORLD;
+        _updateAligns: function() {
+            var wd = this._element.width - this.width;
+            var hd = this._element.height - this.height;
+
+            if (this._align == pc.TEXT_ALIGN_CENTER) {
```

Comment:
We do use two line condition annotation sometimes, where it really shortens number of lines, like in this method:
```javascript
if (this._align == pc.TEXT_ALIGN_CENTER)
    wd *= 0.5;
```